air-quality-backend/src/air_quality/etl/forecast/forecast_data.py

```diff
@@ -134,3 +136,31 @@ def get_valid_time_values(self):
 
     def get_time_value(self) -> int:
         return int(self._single_level_data["time"].values)
+
+    def get_surface_pressure(
+        self, longitude: int, latitude: int, forecast_datetime: datetime
+    ):
+        if self._cached_pressure is None:
+            single_pres = self._single_level_data["sp"]
+            self._cached_pressure = single_pres.set_index(step=["valid_time"])
+
+        since_epoch = forecast_datetime.timestamp()
+
+        single_point = self._cached_pressure.sel(
+            longitude=longitude, latitude=latitude, step=since_epoch, method="nearest"
+        )
```

Comment:
What's the runtime like with his?

This is where we got a chunk of time back when looking up pollutant data for each city before. It's possible to pass in the full list of longitude and latitudes in one call and get back surface pressures for all coordinates we need (same for temperature).

So it would mean we only call the .sel function once for all stations basically. We'd just need to associated the output with each lat/long

The code isn't as nice that way imo but I think it will massively improve performance.

Reply:
It's about 40s for the transform, which isn't ideal. I'd like to merge this in and then sort that in a separate PR if you agree, as I think it might be a little involved